pytato/array.py

```diff
@@ -1804,4 +1810,155 @@ def ones(namespace: Namespace, shape: ConvertibleToShape, dtype: Any = float,
 
 # }}}
 
+
+# {{{ make_index_lambda
+
+INDEX_RE = re.compile("_r?(0|([1-9][0-9]*))")
+
+
+def make_index_lambda(namespace: Namespace,
+        expression: Union[str, ScalarExpression],
+        bindings: Dict[str, Array],
+        shape: ShapeType,
+        dtype: Any) -> IndexLambda:
+    if isinstance(expression, str):
+        raise NotImplementedError("Sorry the developers were too lazy to implement"
+                " a parser.")
```

Comment:
```suggestion
        raise NotImplementedError
```
(no need to apologize)